### PR TITLE
Allow to disable file rooms

### LINKED ISSUE #2280: Is it possible to deactivate talk on file sharing web pages?
Guys I noticed that when I generate a file sharing scream, whatever it is, opening the url in the page also gives me the possibility to call as an anonymous user. in practice it enables talks in the pages of the url that I generate for sharing.

How can I turn off talk in the browser that points to shared urls?

I attach photo

Immagine

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Depends on #2281 - only the last commit is new.

> Bildschirmfoto von 2019-10-08 11-49-00

Fix #2280

### CODE REVIEW

File: lib/Controller/FilesIntegrationController.php [hunk 1]
```diff
@@ -112,6 +117,10 @@ public function __construct(
 	 * @throws OCSNotFoundException
 	 */
 	public function getRoomByFileId(string $fileId): DataResponse {
+		if ($this->config->getAppValue('spreed', 'conversations_files', '1') !== '1') {
+			throw new OCSException($this->l->t('File is not shared, or shared but not with the user'), Http::STATUS_UNAUTHORIZED);
```
Comment (danxuliu) [inline lib/Controller/FilesIntegrationController.php]: The exception message does not match the reason why it was thrown.

File: lib/Files/TemplateLoader.php [hunk 2]
```diff
@@ -33,7 +33,9 @@ class TemplateLoader {
 
 	public static function register(EventDispatcherInterface $dispatcher): void {
 		$dispatcher->addListener('OCA\Files::loadAdditionalScripts', function() {
-			self::loadTalkSidebarForFilesApp();
+			if (\OC::$server->getConfig()->getAppValue('spreed', 'conversations_files', '1') === '1') {
+				self::loadTalkSidebarForFilesApp();
+			}
```
Comment (danxuliu) [inline lib/Files/TemplateLoader.php]: I would always call `self::loadTalkSidebarForFilesApp()` and then return without adding the style and script inside the function if the setting is not enabled, like done in [the TemplateLoader for the public share auth](https://github.com/nextcloud/spreed/blob/d4fc586f1a796b8cb19cff543b668470556e44b2/lib/PublicShareAuth/TemplateLoader.php).

File: lib/PublicShare/TemplateLoader.php [hunk 3]
```diff
@@ -38,7 +38,11 @@ class TemplateLoader {
 
 	public static function register(EventDispatcherInterface $dispatcher): void {
 		$listener = function() {
-			self::loadTalkSidebarUi();
+			$config = \OC::$server->getConfig();
+			if ($config->getAppValue('spreed', 'conversations_files', '1') === '1' &&
+				$config->getAppValue('spreed', 'conversations_files_public_shares', '1') === '1') {
+				self::loadTalkSidebarUi();
+			}
```
Comment (danxuliu) [inline lib/PublicShare/TemplateLoader.php]: Same as above.

File: lib/Controller/FilesIntegrationController.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Controller/FilesIntegrationController.php]: yeah, it's the same for all cases so I just picked the same. I guess this is to not reveal if a file id exists and everything. So we could do a different one here, but not sure it's too useful

**Comment by nickvergessen** [comment]
Changed all of them

**Comment by nickvergessen** [comment]
Rebased after #2331

**Comment by thevril** [comment]
does this feature still exist?

I've got an account on https://cloud.disroot.org 
But I can't find this setting anywhere. The chat is active for every file I share.
I've tried to look for this setting also in a newer cloud with Nextcloud 25, but I didn't find anything!

Can someone write me in detail where is this setting? 
Is this a setting for normal users or only for those who manage the server?

thank in advance!

**Comment by nickvergessen** [comment]
Admin settings

**Comment by thevril** [comment]
> Admin settings

I'm sorry, but is like one hour that I'm searching this.


Cattura2

I do not find here this setting, what am I doing wrong?
The problem is that I'm not admin of the server?

**Comment by nickvergessen** [comment]
> The problem is that I'm not admin of the server?

Yes

**Pull request merged**